### PR TITLE
Add compiler function for Vyper

### PR DESCRIPTION
**Pull request by DanglSan**

### CODE REVIEW

File: smart-contract-verifier/src/vyper/compiler.rs [hunk 1]
```diff
@@ -0,0 +1,544 @@
...
+use std::process::{Stdio, Command};
+use std::str::FromStr;
+use ethers_solc::artifacts::{Ast, Contract, NodeType, Severity, output_selection::OutputSelection};
+use serde::{Deserialize, Serialize};
+use colored::Colorize;
+
+use super::errors::VyperError;
+
+pub type Sources = BTreeMap<PathBuf, Source>;
+pub type FileToContractsMap<T> = BTreeMap<String, BTreeMap<String, T>>;
```
Comment (leviska) [inline smart-contract-verifier/src/vyper/compiler.rs:13]: why is it generic?

File: smart-contract-verifier/src/vyper/compiler.rs [hunk 2]
```diff
@@ -0,0 +1,544 @@
...
+//             } else {
+//                 self
+//             })
+//         } else {
+//             None
+//         }
+//     }
+// }
+
+impl fmt::Display for EvmVersion {
```
Comment (leviska) [inline smart-contract-verifier/src/vyper/compiler.rs:64]: may be use macro https://docs.rs/strum/latest/strum/derive.Display.html

File: smart-contract-verifier/src/vyper/compiler.rs [hunk 3]
```diff
@@ -0,0 +1,544 @@
...
+            EvmVersion::Byzantium => "byzantium",
+            EvmVersion::Constantinople => "constantinople",
+            EvmVersion::Petersburg => "petersburg",
+            EvmVersion::Istanbul => "istanbul",
+        };
+        write!(f, "{}", string)
+    }
+}
+
+impl FromStr for EvmVersion {
```
Comment (leviska) [inline smart-contract-verifier/src/vyper/compiler.rs:76]: may be use https://docs.rs/strum/latest/strum/derive.EnumString.html

File: smart-contract-verifier/src/vyper/compiler.rs [hunk 4]
```diff
@@ -0,0 +1,544 @@
...
+    skip_serializing_if = "Option::is_none")]
+    pub evm_version: Option<EvmVersion>,
+    pub optimize: bool,
+    #[serde(default)]
+    pub output_selection: OutputSelection,
+}
+
+impl Settings {
+    /// Creates a new `Settings` instance with the given `output_selection`
+    pub fn new(output_selection: impl Into<OutputSelection>) -> Self {
```
Comment (leviska) [inline smart-contract-verifier/src/vyper/compiler.rs:155]: you can just accept `OutputSelection`, and call `.into()` when you're calling the function, like
```Settings::new(foo.into())```

File: smart-contract-verifier/src/vyper/compiler.rs [hunk 5]
```diff
@@ -0,0 +1,544 @@
...
+    // pub fn push_output_selection(&mut self, value: impl ToString) {
+    //     self.push_contract_output_selection("*", value)
+    // }
+
+    /// Inserts the `key` `value` pair to the `output_selection` for all files
+    ///
+    /// If the `key` already exists, then the value is added to the existing list
+    pub fn push_contract_output_selection(
+        &mut self,
+        contracts: impl Into<String>,
```
Comment (leviska) [inline smart-contract-verifier/src/vyper/compiler.rs:195]: again, you can just accept string

File: smart-contract-verifier/src/vyper/compiler.rs [hunk 6]
```diff
@@ -0,0 +1,544 @@
...
+        self.set_contract_output_selection("*", values)
+    }
+
+    /// Sets the `key` to the `values` pair to the `output_selection` for all files
+    ///
+    /// This will replace the existing values for `key` if they're present
+    pub fn set_contract_output_selection(
+        &mut self,
+        key: impl Into<String>,
+        values: impl IntoIterator<Item = impl ToString>,
```
Comment (leviska) [inline smart-contract-verifier/src/vyper/compiler.rs:222]: you can accept `impl IntoIterator<Item = String>`

**Comment by rimrakhimov** [comment]
Implemented in #99

**Pull request closed without merging**